store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/handler/DeleteAllDataHandler.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.accumulostore.operation.handler;
+
+import org.apache.accumulo.core.client.Connector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.gov.gchq.gaffer.accumulostore.AccumuloProperties;
+import uk.gov.gchq.gaffer.core.exception.GafferCheckedException;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.operation.DeleteAllData;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+
+import static uk.gov.gchq.gaffer.accumulostore.utils.TableUtils.getConnector;
+
+public class DeleteAllDataHandler implements OperationHandler<DeleteAllData> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeleteAllDataHandler.class);
+
+
+    @Override
+    public Object doOperation(final DeleteAllData operation, final Context context, final Store store) throws OperationException {
+        try {
+            final String removeId = store.getGraphId();
+            try {
+                Connector connection = getConnector((AccumuloProperties) store.getProperties());
+                if (connection.tableOperations().exists(removeId)) {
+                    connection.tableOperations().offline(removeId);
+                    connection.tableOperations().delete(removeId);
```

Comment:
I don't see any test/ITs for this.

Reply:
I don't think it  be possible to test this easily. 
An accumulo instance will need to be set up. 
This will need to be manually tested by developers.

Reply:
Should be possible to test with the MiniAccumuloStore as used by other tests. This doesn't require a real/full Accumulo instance/cluster.

Reply:
This test does no work.
the last assert throws a null pointer.
I do not think testing is easily possible.

```java
FederatedStoreTestUtil.resetForFederatedTests();

//Make a FederatedStore
final FederatedStore federatedStore = new FederatedStore();
federatedStore.initialise(GRAPH_ID_TEST_FEDERATED_STORE, new Schema(), new FederatedStoreProperties());

//Add MiniAccumulo
federatedStore.execute(new AddGraph.Builder()
.graphId(DELETE_ALL_DATA_GRAPH)
.schema(loadSchemaFromJson(SCHEMA_EDGE_BASIC_JSON))
.storeProperties(loadAccumuloStoreProperties(PROPERTIES_ACCUMULO_STORE_PROPERTIES))
.build(), contextBlankUser());

//Make a Local MiniAccumulo with Table collision.
final Store directToMiniAccumulo = MiniAccumuloStore.createStore(DELETE_ALL_DATA_GRAPH, loadSchemaFromJson(SCHEMA_EDGE_BASIC_JSON), loadAccumuloStoreProperties(PROPERTIES_ACCUMULO_STORE_PROPERTIES));

//Local Mini is empty
assertThat(directToMiniAccumulo.execute(new GetAllElements(), contextBlankUser()))
.containsExactly();

//Add Element to FederatedStore
federatedStore.execute(new AddElements.Builder().input(edgeBasic()).build(), contextBlankUser());

//Check the element is Local MiniAccumulo confirming table collision.
assertThat(directToMiniAccumulo.execute(new GetAllElements(), contextBlankUser()))
.containsExactly(edgeBasic());

//Federated RemoveGraph with no Deletion of data or tables
federatedStore.execute(new RemoveGraph.Builder().graphId(DELETE_ALL_DATA_GRAPH).build(), contextBlankUser());

//Confirm Local mini is still working and connected to the table.
assertThatNoException()
.isThrownBy(() -> directToMiniAccumulo.execute(new GetAllElements(), contextBlankUser()));
```